lib/improver/blending/weighted_blend.py

```diff
@@ -353,6 +432,21 @@ def process(self, cube, weights=None):
             result = cube.collapsed(self.coord,
                                     iris.analysis.MEAN, weights=weights_array)
 
+        # Else use the maximum probability aggregator.
+        elif self.mode == "weighted_maximum":
```

Comment:
Caroline did want the code in each if block to be a separate function. I could do this, but each one only has 2 or 3 lines of code. Suggestions welcome.